nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/CopyAzureBlobStorage_v12.java

```diff
@@ -0,0 +1,388 @@
+
+                AzureStorageCredentialsDetails_v12 credentialsDetails = fromCredentialsService.getCredentialsDetails(flowFile.getAttributes());
+                String sourceUrl = fromBlobClient.getBlobUrl();
+                final BlobProperties fromBlobProperties = fromBlobClient.getProperties();
+                final long blobSize = fromBlobProperties.getBlobSize();
+
+                final BlobRequestConditions sourceRequestConditions = new BlobRequestConditions();
+                sourceRequestConditions.setIfMatch(fromBlobProperties.getETag());
+
+                HttpAuthorization httpAuthorization;
+                final String sasToken = (credentialsDetails.getCredentialsType() == AzureStorageCredentialsType.ACCOUNT_KEY)
+                        ? generateSas(fromContainerClient)
+                        : credentialsDetails.getSasToken();
+                if (sasToken != null) {
+                    sourceUrl += "?" + sasToken;
+                    httpAuthorization = null;
+                } else {
+                    httpAuthorization = getHttpAuthorization(credentialsDetails);
+                }
+
+                copy(blobClient, httpAuthorization, sourceUrl, blobSize, sourceRequestConditions, destinationRequestConditions);
+                applyBlobMetadata(attributes, blobClient);
+
+                if (ignore) {
+                    attributes.put(ATTR_NAME_IGNORED, "false");
+                }
+            } catch (BlobStorageException e) {
+                final BlobErrorCode errorCode = e.getErrorCode();
+                flowFile = session.putAttribute(flowFile, ATTR_NAME_ERROR_CODE, e.getErrorCode().toString());
+
+                if (errorCode == BlobErrorCode.BLOB_ALREADY_EXISTS && ignore) {
+                    getLogger().info("Blob already exists: remote blob not modified. Transferring {} to success", flowFile);
+                    attributes.put(ATTR_NAME_IGNORED, "true");
+                } else {
+                    throw e;
+                }
+            }
+
+            flowFile = session.putAllAttributes(flowFile, attributes);
+            session.transfer(flowFile, REL_SUCCESS);
+
+            long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+            String transitUri = attributes.get(ATTR_NAME_PRIMARY_URI);
+            session.getProvenanceReporter().send(flowFile, transitUri, transferMillis);
+        } catch (Exception e) {
+            getLogger().error("Failed to create blob on Azure Blob Storage", e);
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+        }
+    }
+
+    private static String generateSas(final BlobContainerClient fromContainerClient) {
+        BlobContainerSasPermission permissions = new BlobContainerSasPermission().setCreatePermission(true).setWritePermission(true).setAddPermission(true).setReadPermission(true);
+        OffsetDateTime utc = OffsetDateTime.now(ZoneOffset.UTC);
```

Comment:
If UTC is required, it would be worth adding a comment here to indicate that UTC is necessary for proper signature generation.